test/new-e2e/tests/ha-agent/haagent_metadata_test.go

```diff
@@ -0,0 +1,49 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2025-present Datadog, Inc.
+
+package haagent
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/DataDog/test-infra-definitions/components/datadog/agentparams"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/e2e"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/environments"
+	awshost "github.com/DataDog/datadog-agent/test/new-e2e/pkg/provisioners/aws/host"
+)
+
+type haAgentMetadataTestSuite struct {
+	e2e.BaseSuite[environments.Host]
+}
+
+// TestHaAgentMetadataSuite runs the HA Agent Metadata e2e suite
+func TestHaAgentMetadataSuite(t *testing.T) {
+	// language=yaml
+	agentConfig := `
+ha_agent:
+    enabled: true
+config_id: test-config01
+log_level: debug
+`
+	e2e.Run(t, &haAgentMetadataTestSuite{}, e2e.WithProvisioner(awshost.Provisioner(
+		awshost.WithAgentOptions(agentparams.WithAgentConfig(agentConfig))),
+	))
+}
```

Comment:
Ideally, for this test, I think we should avoid relying on real backend, and just check that `show-metadata ha-agent` CLI works with unknown state.